test/sandbox/test/inline_template_test.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class InlineErbTest < ViewComponent::TestCase
+  class InlineErbComponent < ViewComponent::Base
+    include ViewComponent::InlineTemplate
+
+    attr_reader :name
+
+    template! <<~ERB
+      <h1>Hello, <%= name %>!</h1>
+    ERB
+
+    def initialize(name)
+      @name = name
+    end
+  end
+
+  class InlineRaiseErbComponent < ViewComponent::Base
+    include ViewComponent::InlineTemplate
+
+    attr_reader :name
+
+    template! <<~ERB
```

Comment:
Could we register methods for each valid handler?

```suggestion
   erb_template! <<~ERB
```

Reply:
I'm hacking on this idea, but I have some concerns that some environments may run into load order problems. To do this in an efficient way, we'll likely need to do something like:

```ruby
module InlineTemplates
  class_methods do
    ActionView::Template.template_handler_extensions.each do |ext|
      define_method("#{ext}_template!") do 
      end
    end
end
```

In our office hours we discussed that this would be similar to the compiler, but I think it's slightly different because this has to happen when the *module* is first evaluated/included because these are class methods (or use the slower `method_missing` approach). When we compile, we explicitly call `compile` on all descendants after `ActiveSupport` is loaded, which gives us some control over what's loaded at the time vs it being when it's first required/evaluated.

I don't have time to setup an environment to try confirming that, but wanted to throw that out there as a potential gotcha.

Reply:
Okay, I worked around that with some clever use of `method_missing`. It feels more Ruby/Rails like since it's a bit more magical. 😄